x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/cohere/CohereService.java

```diff
@@ -293,6 +300,18 @@ private CohereEmbeddingsModel updateModelWithEmbeddingDetails(CohereEmbeddingsMo
         return new CohereEmbeddingsModel(model, serviceSettings);
     }
 
+    static SimilarityMeasure defaultSimilarity(CohereEmbeddingType embeddingType) {
+        if (embeddingType == null) {
+            return SimilarityMeasure.DOT_PRODUCT;
+        }
+
+        return switch (embeddingType) {
+            case FLOAT -> SimilarityMeasure.DOT_PRODUCT;
+            case BYTE -> SimilarityMeasure.COSINE;
+            case INT8 -> SimilarityMeasure.COSINE;
+        };
```

Comment:
For future reference, how did we pick this mapping?

Reply:
Also, maybe we should do some logging to indicate that a default similarity was used?

Reply:
I added some java docs explaining the logic.

For byte embeddings I found they were rejected when  `similarity: dot_product` as ES complained the vector was not normalised. This shouldn't be the case, I'm investigating why and will follow up. I think the PR can be merged anyway for the purpose of unblocking the testing.